### PR TITLE
Add overview page to be shown on startup

### LINKED ISSUE #9: Update extension pack to install core MP extensions only
Consider that the Extension Pack will install core MP extensions only with welcome page pointing to additional extensions that users can choose to install themselves e.g. the MP runtime extensions.

### PR DESCRIPTION
**Pull request by kathrynkodama**
Removes runtime specific extensions from the extension pack and instead recommends them through the new Overview page.

This is how the Overview page will look on installation.

- Selecting either of the commands under the "Start" section will execute the corresponding starter command, ie. selecting "Create a MicroProfile project..." will first check that the MP Starter extension is installed (if not, will prompt user to install) and then call the `extension.microprofileStarter` command.
- "MicroProfile" under "Documentation" links to https://microprofile.io/
- "MicroProfile Runtimes" includes links to runtime specific VS Code extensions, as well as prompts to install each extension.
- "Questions & Issues" under "Help" links to https://github.com/MicroShed/vscode-microprofile-pack/issues

If a user installs a runtime specific extension (ie. Open Liberty Tools is installed), the corresponding "Install ..." prompt will no longer appear in the overview page. 

If for any reason, one of the extensions included in the extension pack is uninstalled, the Overview page will have a new section titled "Extensions" that will prompt the user to re-install the corresponding extension.

Fixes #9 
Signed-off-by: Kathryn Kodama <kathryn.s.kodama@gmail.com>

### CODE REVIEW

**Comment by Emily-Jiang** [comment]
@kathrynkodama what about the lsp4mp extensions? Can you list them under the MP pack?

**Comment by kathrynkodama** [comment]
> @kathrynkodama what about the lsp4mp extensions? Can you list them under the MP pack?

The lsp4mp extension is "Tools for MicroProfile" and is already included in the MP pack.

File: package.json [hunk 1]
```diff
@@ -33,12 +33,68 @@
   "bugs": {
     "url": "https://github.com/MicroShed/vscode-microprofile-pack/issues"
   },
+  "main": "./out/extension.js",
+  "activationEvents": [
+    "onStartupFinished",
```
[the remark this answers is not in the document]
Comment (kathrynkodama) [reply package.json:38]: I was thinking we would want to show the Overview page even if there is not a project open in the current window, as then they could follow the prompt to generate a new MicroProfile starter project.  I am open to changing it to activating on java projects if showing the overview page on every new instance of VS Code is too much.

[the remark this answers is not in the document]
Comment (rzgry) [reply package.json:38]: Only thing that i'm concerned about is if they open a Python/JS project or some other language. Seems a little strange the MicroProfile overview page would open.

[the remark this answers is not in the document]
Comment (kathrynkodama) [reply package.json]: Good point, updated this to only start up on Java projects.

**Comment by Emily-Jiang** [comment]
> @kathrynkodama what about the lsp4mp extensions? Can you list them under the MP pack?

cool, thanks!

File: src/overview/overviewPage.ts [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+    // monitor extensions installed
+    vscode.extensions.onDidChange(e => {
+      syncExtensionVisibility(this._panelView);
+    });
+
+    // monitor workspace configuration for alwaysShowOverview setting
+    vscode.workspace.onDidChangeConfiguration((event: vscode.ConfigurationChangeEvent) => {
+      if (event.affectsConfiguration(MICROPROFILE_SHOW_OVERVIEW_CONFIGURATION)) {
+        syncCheckboxValue(this._panelView);
+      }
```
Comment (rzgry) [inline src/overview/overviewPage.ts:91]: Can probably add these subscriptions to context.subscriptions so they are removed if extension is uninstalled/disabled.

**Review by rzgry — APPROVED** [review]

**Review by ericglau — APPROVED** [review]

**Comment by kathrynkodama** [comment]
Updated the documentation section to include the following links:
- Landing Page http://microprofile.io/
- Learning Resources https://wiki.eclipse.org/MicroProfile/Learning_Resources
- FAQ https://microprofile.io/faq/
- MicroProfile blog http://microprofile.io/blog

image

**Review by rzgry — APPROVED** [review]

**Pull request merged**